tests/test_cases/test_discovery/test_discovery.py

```diff
@@ -471,3 +471,17 @@ def type_check_verilog(dut):
         tlog.info("Handle %s" %  (handle[0]._fullname,))
         if handle[0]._type != handle[1]:
             raise TestFailure("Expected %s found %s for %s" % (handle[1], handle[0]._type, handle[0]._fullname))
+
+
+@cocotb.test()
```

Comment:
This should be marked verilog-only - underscores are illegal at the start of VHDL identifiers

Reply:
True, I had forgotten to add that.
But since it's valid using extended identifiers, I maybe should wait until your #1610 is in and add this test near it?